### DISCUSSION TITLE
`clusterProperties` Expressions behave different on iOS

### DISCUSSION THREAD

**Discussion by jblemee** [maplibre/maplibre-react-native#569] [Q&A] [answered]
### Steps to Trigger Behavior

1. Create a clustered `ShapeSource`
```
 <ShapeSource
        id="map-opportunities"
        ref={shapeSourceRef}
        cluster
        clusterRadius={50}
        clusterMaxZoomLevel={14}
        clusterProperties={{
            // does not work on ios
            maxPrice: ['max', ['get', 'price']],
        }}
        shape={geojsonFeed}
    
    >
```
2. Try to use the cluster property on a `SymbolLayer`

```
 <SymbolLayer
        id="priceLayer"
        style={{
          ...
            textField: ['get', 'maxPrice'],
        ...
        }}
       ...
    />
```
3.

### Expected Behavior
Show the max price **number**

### Actual Behavior
It shows the **string** `max` on IOS (both simulator and real device). It works on android (both emulator and real device)

### Screenshots (if applicable)

![Image](https://github.com/user-attachments/assets/8df5c7f4-17ee-47fb-b07f-c756aacce8bc)

### Environment

- Platform: IOS
- OS version: iOS 18
- Device type: iPhone 13 and simulator
- Emulator or Simulator: yes
- Development OS: macOS 15
- `@maplibre/maplibre-react-native` Version: 10.0.0-beta.8 and 10.0.0-beta.11 (at least)
- MapLibre Native Version: do not know
- `react-native` Version: 0.76.3 (with newArch)
- `expo` Version: no expo

**Comment by KiwiKilian** [comment]
Did you have a look at the [earth quakes example](https://github.com/maplibre/maplibre-react-native/blob/a130786d8b036d1f9700774fba031f408364194e/packages/examples/src/examples/SymbolCircleLayer/EarthQuakes.tsx#L209-L230)?

Can you try if this works?

```
        clusterProperties={{
            maxPrice: ['max', 'accumulated', ['get', 'price']],
        }}
```

Maybe even with `['accumulated']`?

Otherwise please create a minimal example within this file: https://github.com/maplibre/maplibre-react-native/blob/beta/packages/examples/src/examples/BugReport.tsx

**Comment by jblemee** [comment]
> Did you have a look at the [earth quakes example](https://github.com/maplibre/maplibre-react-native/blob/a130786d8b036d1f9700774fba031f408364194e/packages/examples/src/examples/SymbolCircleLayer/EarthQuakes.tsx#L209-L230)?
> 

Of course

According to the documentation (https://docs.mapbox.com/style-spec/reference/sources/#geojson-clusterProperties), the 'accumulated' syntax is optional.

> Otherwise please create a minimal example within this file: https://github.com/maplibre/maplibre-react-native/blob/beta/packages/examples/src/examples/BugReport.tsx
>

I'll do it if the "accumulated" can't be used as a workaround

**Comment by jblemee** [comment, marked as answer]
Ok, it works with the "accumulated" syntax and it also works with the short syntax..

The issue was that the 'short' syntax failed on IOS when there is some `undefined`numbers. Seems like Android is converting them to 0 but IOS do not

**Comment by KiwiKilian** [comment]
I guess this might also be solvable with expressions, to fallback to zero? Might be worth to report this back to [MapLibre Native](https://github.com/maplibre/maplibre-native)? Closing here, as the RN wrapper isn't at fault here.
